src/chat_events.cpp

```diff
@@ -127,6 +128,11 @@ void chat_handler::do_speak(const std::string& message, bool allies_only)
 	cch.dispatch(cmd);
 }
 
+void chat_handler::clear()
+{
+	game_display::get_singleton()->get_chat_manager().clear_chat_messages();
```

Comment:
This class didn't depend on game_display before, and AFAICS there's nothing else that indicates a game_display instance exists. I'd suggest making chat_handler::clear() be a pure virtual function, and then only using game_display in subclasses that already use game_display.

Reply:
Gotcha. So I moved the `menu_handler::clear_messages` definition to `chat_handler` and made it a pure virtual.

Reply:
Okay, there are testing failing. I'll fix it later

Reply:
@stevecotton the test is fixed now